File: study/ts/script/scene/view/settlement/SummarySeasonEnd.ts
import SummaryBase from "./SummaryBase";
import { G_EffectGfxMgr, G_ResolutionManager, G_UserData, G_SceneManager } from "../../../init";
import { handler } from "../../../utils/handler";
import ComponentBase from "./ComponentBase";
import ComponentSeasonPlayers from "./ComponentSeasonPlayers";

const { ccclass, property } = cc._decorator;
@ccclass
export default class SummarySeasonEnd extends SummaryBase {

    private _battleData;
    public init(battleData, callback, attackHurt?) {
        this._battleData = battleData;
        var list: ComponentBase[] = [];

        var height = G_ResolutionManager.getDesignHeight();

        let midXPos = 0;
        var componentSeasonPlayers = new cc.Node("componentSeasonPlayers").addComponent(ComponentSeasonPlayers);
        componentSeasonPlayers.init(battleData, cc.v2(midXPos, 265 - height * 0.5))
        list.push(componentSeasonPlayers);

        super.init(this._battleData, callback, list);
    }
    public onEnter() {
        super.onEnter();
        this._createAnimation();
    }
    public onExit() {
        if (G_UserData.getSeasonSport().isSquadReconnect() && G_UserData.getSeasonSport().isPlayReport() == false) {
            G_SceneManager.showScene("seasonSport")
        }

        G_UserData.getSeasonSport().setPlayReport(false)
        super.onExit();
    }
    private _createAnimation() {
        G_EffectGfxMgr.createPlayMovingGfx(this.node, 'moving_battleend', null, handler(this, this.checkStart), false);
    }
}

File: study/lua/src/app/scene/view/settlement/SummarySeasonEnd.lua
-- -- 无差别战斗结束

local SummaryBase = require("app.scene.view.settlement.SummaryBase")
local SummarySeasonEnd = class("SummarySeasonEnd", SummaryBase)

function SummarySeasonEnd:ctor(battleData, callback, attackHurt)
    self._battleData = battleData
    local list = {}
    local size = G_ResolutionManager:getDesignCCSize()
    local width = size.width
    local height = size.height

    local midXPos = 0

    local ComponentSeasonPlayers = require("app.scene.view.settlement.ComponentSeasonPlayers").new(battleData, cc.p(midXPos, 265 - height*0.5))
    table.insert(list, ComponentSeasonPlayers)

    SummarySeasonEnd.super.ctor(self, battleData, callback, list)
end

function SummarySeasonEnd:onEnter()
    SummarySeasonEnd.super.onEnter(self)
    self:_createAnimation()
end

function SummarySeasonEnd:onExit()
    if G_UserData:getSeasonSport():isSquadReconnect() and G_UserData:getSeasonSport():isPlayReport() == false then
        G_SceneManager:showScene("seasonSport")
    end
    
    G_UserData:getSeasonSport():setPlayReport(false)
    SummarySeasonEnd.super.onExit(self)
end

function SummarySeasonEnd:_createAnimation()
    local EffectGfxNode = require("app.effect.EffectGfxNode")
    local function effectFunction(effect)
        local subEffect = EffectGfxNode.new(effect)
        subEffect:play()
        return subEffect
    end
    G_EffectGfxMgr:createPlayMovingGfx( self, "moving_battleend", effectFunction, handler(self, self.checkStart) , false )
end

return SummarySeasonEnd